site/src/pages/GroupsPage/GroupPage.tsx

```diff
@@ -146,7 +147,13 @@ export const GroupPage: React.FC = () => {
             </PageHeader>
 
             <Stack spacing={1}>
-              <Maybe condition={canUpdateGroup}>
+              <Maybe
+                condition={
+                  canUpdateGroup &&
+                  group !== undefined &&
+                  group?.id !== group?.organization_id
```

Comment:
should have another helper here to check for everyone  group... new devs are not going to understand this code as written